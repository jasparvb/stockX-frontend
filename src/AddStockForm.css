.AddStockForm {
  position: absolute;
  background: rgb(0 0 0 / 0.3);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 10;
}

.AddStockForm .form-container {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
}

.AddStockForm .form-container .form-column {
  box-shadow: 0px 1px 10px 0px #6b6b6b;
  max-width: 450px;
  border: 1px solid #ced4da;
  border-radius: 3px;
  background: white;
  padding: 20px;
}

