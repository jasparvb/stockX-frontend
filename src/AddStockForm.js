import React, { useEffect } from "react";
import { useDispatch } from 'react-redux';
import { addNewStockAPI } from './actions/lists';
import './AddStockForm.css';
import { useClickOutside } from 'react-click-outside-hook';
import AddListForm from './AddListForm';

function AddStockForm({ticker, name, setFormVisible, lists}) {
  const dispatch = useDispatch();

  const [ref, hasClickedOutside] = useClickOutside();

  useEffect(() => {
    if(hasClickedOutside) {
      setFormVisible(false);
    }
  },[hasClickedOutside]);

  function addStock(listId) {
    dispatch(addNewStockAPI(ticker, name, listId));
    setFormVisible(false);
  }

  return (
    <div className="AddStockForm container-fluid text-left">
      <div className="overlay"></div>
      <div className="form-container">
        <div ref={ref} className="form-column">
          <h3 className="mb-3">Create a new list</h3>
          <AddListForm />
          {lists.length ?
            <>
            <p className="mt-4">Add stock to a list below</p>
            {lists.map(l => (
              <button 
              key={l.id} 
              className="btn btn-outline-primary" 
              onClick={() => addStock(l.id)}>Add to "{l.name}"
              </button>
            ))}
            </>
            : ''
          }
        </div>
      </div>
    </div>
  );
}

export default AddStockForm;