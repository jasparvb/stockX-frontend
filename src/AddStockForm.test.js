import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from "react-redux";
import rootReducer from "./reducers/root";
import { BrowserRouter } from "react-router-dom";
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from 'redux-devtools-extension';
import AddStockForm from './AddStockForm';

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)
  )
);

const lists = [
  {
    id: 1,
    name: "Tech Stocks"
  }
]

it("renders without crashing", function() {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <AddStockForm lists={lists} />
      </BrowserRouter>
    </Provider>);
});

it("matches snapshot", function() {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <AddStockForm lists={lists} />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});