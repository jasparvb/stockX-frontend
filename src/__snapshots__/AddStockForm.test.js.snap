// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="AddStockForm container-fluid text-left"
  >
    <div
      class="overlay"
    />
    <div
      class="form-container"
    >
      <div
        class="form-column"
      >
        <h3
          class="mb-3"
        >
          Create a new list
        </h3>
        <div
          class="AddListForm"
        >
          <form>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                name="title"
                placeholder="New list title"
                type="text"
                value=""
              />
            </div>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Add
            </button>
          </form>
        </div>
        <p
          class="mt-4"
        >
          Add stock to a list below
        </p>
        <button
          class="btn btn-outline-primary"
        >
          Add to "Tech Stocks"
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
